ibis/backends/pyspark/compiler.py

```diff
@@ -1655,6 +1655,30 @@ def compile_array_collect(t, op, **kwargs):
     return F.collect_list(src_column)
 
 
+PARAM_ARG_COLUMNS = 'argColumns'
+
+
+@compiles(ops.Argument)
+def compile_argument(t, op, **kwargs):
+    if PARAM_ARG_COLUMNS in kwargs:
+        return kwargs[PARAM_ARG_COLUMNS][op.name]
+    else:
+        raise ValueError(
+            f'{PARAM_ARG_COLUMNS} needs to be passed for ops.Argument to be translated'
+        )
```

Comment:
Any reason we can't make this an explicit keyword argument?

```suggestion


@compiles(ops.Argument)
def compile_argument(t, op, arg_columns=None, **kwargs):
    if arg_columns is not None:
        return arg_columns[op.name]
    else:
        raise ValueError(
            f'`arg_columns` needs to be passed for ops.Argument to be translated'
        )
```

Reply:
No reason. I instinctively thought it wouldn't be passed down the chain this way, but of course it will. In this case I should probably also make arg_columns a required parameter and drop the validation as well.

Reply:
No worries!

If you're interested, after this PR you should be able to use the same approach to implement `ops.ArrayMap`!

Reply:
Made the changes, thanks. Sure, I will add `ops.ArrayMap` as well.